bin/processDragenData.sh

```diff
@@ -0,0 +1,333 @@
+log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Found gsBatchDirs: ${gsBatchDirs[*]:-}"
+
+if [[ "${#gsBatchDirs[@]:-0}" -eq '0' ]]
+then
+	log4Bash 'WARN' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "No batch directories found in ${TMP_ROOT_DIR}/"
+else
+	for gsBatchDir in "${gsBatchDirs[@]}"
+	do
+		#
+		# Process this batch.
+		#
+		gsBatch="$(basename "${gsBatchDir}")"
+		controlFileBase="${TMP_ROOT_DIR}/logs/${gsBatch}/${gsBatch}"
+		export JOB_CONTROLE_FILE_BASE="${controlFileBase}.${SCRIPT_NAME}"
+		#
+		# ToDo: change location of log files back to ${TMP_ROOT_DIR} once we have a 
+		#       proper prm mount on the GD clusters and this script can run a GD cluster
+		#       instead of on a research cluster.
+		#
+		if [[ -e "${JOB_CONTROLE_FILE_BASE}.finished" ]]
+		then
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Skipping already processed batch ${gsBatch}."
+			continue
+		else
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Processing batch ${gsBatch}..."
+		fi
+		# shellcheck disable=SC2174
+		mkdir -m 2770 -p "${TMP_ROOT_DIR}/logs/"
+		# shellcheck disable=SC2174
+		mkdir -m 2770 -p "${TMP_ROOT_DIR}/logs/${gsBatch}/"
+		printf '' > "${JOB_CONTROLE_FILE_BASE}.started"
+	
+		if [[ -e "${TMP_ROOT_DIR}/${gsBatch}/${gsBatch}.finished" ]]
+		then
+			log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "${TMP_ROOT_DIR}/${gsBatch}/${gsBatch}.finished present -> Data transfer completed; let's process batch ${gsBatch}..."
+			_sampleSheet="GS_DRAGEN_1.csv"
+			#
+			# Get fields (columns) from samplesheet.
+			#
+			declare -a sampleSheetColumnNames=()
+			declare -A sampleSheetColumnOffsets=()
+			IFS="," read -r -a sampleSheetColumnNames <<< "$(head -1 "${sampleSheet}")"
+			for (( _offset = 0 ; _offset < ${#sampleSheetColumnNames[@]:-0} ; _offset++ ))
+			do
+			  	sampleSheetColumnOffsets["${sampleSheetColumnNames[${_offset}]}"]="${_offset}"
+			done
+			#### CHANGE externalSampleId check TO sampleProcessStepID!!!
+			if [[ -n "${_sampleSheetColumnOffsets["externalSampleID"]+isset}" ]]
+			then
+				#_sampleProcessStepIDFieldIndex=$((${_sampleSheetColumnOffsets["sampleProcessStepID"]} + 1))
+				#### CHANGE barcode check TO sampleProcessStepID!!!
+				sampleProcessStepIDFieldIndex=$((${sampleSheetColumnOffsets["barcode"]} + 1))
+				externalSampleIDFieldIndex=$((${sampleSheetColumnOffsets["externalSampleID"]} + 1))
+			else
+				echo "does not exist in the header"
+				continue
+			fi
+
+			count='0'
+			while read line
+			do
+				if [[ "${count}" == '0' ]]
+				then
+					count='1'
+				else
+					externalSampleID=$(echo ${line} | cut -d "," -f "${externalSampleIDFieldIndex}")
+					sampleProcessStepID=$(echo ${line} | cut -d "," -f "${sampleProcessStepIDFieldIndex}")
+					sanityChecking "${sampleProcessStepID}" "${gsBatch}" "${controlFileBase}"
+
+
+					renameDragenData ${sampleProcessStepID} ${externalSampleID}
+					echo "${sampleProcessStepID}-${externalSampleID}"
```

Comment:
echo -> log4Bash

Reply:
removed this file for now, since it is still in progress